source/docs/api-reference/api-usage/swerve/swerve-simulation.rst

```diff
@@ -0,0 +1,32 @@
+Swerve Simulation
+=================
+
+The API supports a functionality focused simulation. This means that the simulation API assumes that the swerve drive is perfect (no scrub and no wheel slip). Additionally, it assumes the inertia of the steer module and the drive modules. Robot-wide, rotational and translational inertia is not accounted for.
+
+To update the simulated swerve robot state, ensure ``m_drivetrain.updateSimState(0.02, 12)`` (``m_drivetrain`` is a ``SwerveDrivetrain``, ``0.02`` is the loop update rate, and ``12`` is the battery voltage) is called in ``simulationPeriodic()``.
+
+.. tab-set::
+
+   .. tab-item:: Java
+      :sync: java
+
+      .. code-block:: java
+
+         @Override
+         public void simulationPeriodic() {
+            /* Assume 20ms update rate, 12v battery voltage */
+            updateSimState(0.02, 12);
```

Comment:
This isn't necessary with the `CommandSwerveDrivetrain` because it occurs in the subsystem's `simulationPeriodic` instead.